packages/dart_frog/lib/src/response.dart

```diff
@@ -50,18 +50,34 @@ class Response {
 
   final shelf.Response _response;
 
+  Completer<List<int>>? _bodyBytesCompleter;
+
   /// The HTTP status code of the response.
   int get statusCode => _response.statusCode;
 
   /// The HTTP headers with case-insensitive keys.
   /// The returned map is unmodifiable.
   Map<String, String> get headers => _response.headers;
 
+  Future<List<int>> _bytes() async {
+    if (_bodyBytesCompleter == null) {
+      _bodyBytesCompleter = Completer<List<int>>();
+      final bytes = await _response.read().fold<List<int>>(
+        <int>[],
+        (previous, element) => previous..addAll(element),
+      );
+      _bodyBytesCompleter!.complete(bytes);
+    }
+    return _bodyBytesCompleter!.future;
+  }
+
   /// Returns a [Stream] representing the body.
-  Stream<List<int>> bytes() => _response.read();
```

Comment:
that's the one I wouldn't apply any cache, because if you have 10Gb of data together it is unlikely you want it cached in memory. You'll pipe to filesystem and that's it.

Reply:
That is a good point, maybe we should expose a flag so people can decide if they want it cached or not, and by default make it false? So it wouldn't be a breaking change for anyone that is already doing file transfers?

Reply:
Just pushed some changes so that everything except bytes is cached. Lmk what you think and thanks again for the review!